replacy/__init__.py

```diff
@@ -24,13 +23,38 @@
 )
 from replacy.version import __version__
 
-
 logging.basicConfig(level=logging.INFO)
 
-
 PipelineComponent = Callable[[List[Span]], List[Span]]
 
 
+class ESpan(Span):
+    match_name = None
+    description = None
+    category = None
+    subcategory = None
+    suggestions = []
+    suggestions_separator = None
+    construct_suggestion_function = None
+    score = 0
+    learn_more = ""
+    issue_type = None
+    simple_description = ""
+    paragraph = False
+    header = ""
+    meta_dict = {}
+    accept_all_changes = False
+    debug = False
+    segment = None
+    visible = True
+
+    @staticmethod
+    def create_instance(doc, start, end):
+        span = Span(doc, start, end)
+        espan = ESpan(span.doc, span.start, span.end, span.label, span.vector, span.vector_norm, span.kb_id)
+        return espan
+
+
```

Comment:
this has 2 problems;
1. it makes replaCy more Writer-specific, when we would like to remove anything Writer-specific
2. it doesn't have one of my favorite replaCy features: dynamic properties. if you set a new, unseen key in the match_dict, that becomes a property on the span (well, an extension attribute). so if you make a key in the match_dict called `ooga` and set the value to `"booga"` for only one match, then all other matches will have `match_span._.ooga == ""` but the match which has that key set will have `match_span._.ooga = ="booga"`. For `ESpan` we would like this, but without the underscore

is it possible to instead override `Span.set_extension` to dynamically set these properties? something like:

```python
from spacy.tokens import Span
from spacy.tokens.underscore import get_ext_args


class ESpan(Span):
    @classmethod
    def create_instance(cls, doc, start, end):
        span = Span(doc, start, end)
        espan = cls(span.doc, span.start, span.end, span.label, span.vector, span.vector_norm, span.kb_id)
        return espan

    @classmethod
    def from_span(cls, span: Span):
        # I added this as a helper, we don't need to keep it
        return  cls(span.doc, span.start, span.end, span.label, span.vector, span.vector_norm, span.kb_id)

    @classmethod
    def set_extension(cls, name, **kwargs):
        # if we only want to allow default values, this works:
        default, method, getter, setter = get_ext_args(**kwargs)
        setattr(cls, name, default)
        # if we want to allow getters and setters or methods for dynamic props, we have to implement that
        # I think it is doable using the `property` built-in method as shown here
        # https://stackoverflow.com/a/1355444/3518108
```